sw/host/tests/usbdev/usbdev_stream/usb_device.h

```diff
@@ -0,0 +1,354 @@
+                        libusb_transfer_cb_fn callback, void *user_data,
+                        unsigned timeout_ms) const {
+    libusb_fill_bulk_transfer(xfr, devh_, ep, buffer, len, callback, user_data,
+                              timeout_ms);
+  }
+  /**
+   * Fill Interrupt Transfer descriptor.
+   */
+  void FillIntTransfer(struct libusb_transfer *xfr, uint8_t ep, uint8_t *buffer,
+                       size_t len, libusb_transfer_cb_fn callback,
+                       void *user_data, unsigned timeout_ms) const {
+    libusb_fill_interrupt_transfer(xfr, devh_, ep, buffer, len, callback,
+                                   user_data, timeout_ms);
+  }
+  /**
+   * Fill Isochronous transfer descriptor,
+   */
+  void FillIsoTransfer(struct libusb_transfer *xfr, uint8_t ep, uint8_t *buffer,
+                       size_t len, unsigned num_iso_packets,
+                       libusb_transfer_cb_fn callback, void *user_data,
+                       unsigned timeout_ms) const {
+    libusb_fill_iso_transfer(xfr, devh_, ep, buffer, len, num_iso_packets,
+                             callback, user_data, timeout_ms);
+  }
+  /**
+   * Complete the lengths of the packets in this Isochronous transfer, assuming
+   * that packets are of uniform size.
+   *
+   * @param  xfr     Isochronous transfer to be completed.
+   * @param  len     Total length of all packets within the transfer.
+   */
+  void SetIsoPacketLengths(struct libusb_transfer *xfr, unsigned len) const {
+    libusb_set_iso_packet_lengths(xfr, len);
+  }
+  /**
+   * Allocate a transfer descriptor to be completed by the caller.
+   *
+   * @return  Pointer to transfer transcript, or NULL if allocation failed.
+   */
+  struct libusb_transfer *AllocTransfer(unsigned iso_packets) const {
+    return libusb_alloc_transfer((int)iso_packets);
+  }
+  /**
+   * Free an allocated transfer descriptor.
+   *
+   * @param  xfr     The transfer descriptor to be freed.
+   */
+  void FreeTransfer(struct libusb_transfer *xfr) const {
+    return libusb_free_transfer(xfr);
+  }
+  /**
+   * Submit a transfer for processing.
+   *
+   * @param  xfr     The transfer to be submitted.
+   */
+  int SubmitTransfer(struct libusb_transfer *xfr) const {
+    return libusb_submit_transfer(xfr);
+  }
+  /**
+   * Cancel a pending transfer.
+   *
+   * @param  xfr     The transfer to be cancelled.
+   */
+  int CancelTransfer(struct libusb_transfer *xfr) const {
+    return libusb_cancel_transfer(xfr);
+  }
+#endif
+
+  /**
+   * Read Test Descriptor from the DUT using a Vendor-Specific command.
+   *
+   * @return true iff the operation was successful.
+   */
+  bool ReadTestDesc();
+  /**
+   * Return the test number from the test descriptor.
+   *
+   * @return test number
+   */
+  uint8_t TestNumber() const { return testNumber_; }
+  /**
+   * Return the specified test argument from the test descriptor,
+   *
+   * @param  arg     Argument number.
+   * @return test argument
```

Comment:
Maybe specify somewhere, preferably in the form of a named constant, that there are 4 arguments and explain what they are?

Reply:
I've added a comment. A bit like 'argc/argv', though, it's not really specified and could potentially be test number-dependent. Presently the set of tests remains fairly small.